minetest/minetest
# Docker image workflow (publish on `ghcr.io`)

Should fix https://github.com/minetest/minetest/issues/14164

Follows the official Docker documentation to build image using GitHub Actions workflows.

Official Actions by Docker are used.

Links:
- https://docs.github.com/en/actions/publishing-packages/publishing-docker-images
- https://docs.docker.com/build/ci/github-actions/multi-platform
- https://github.com/opencontainers/image-spec/blob/main/annotations.md

Tags are automatically determined by the Docker metadata Action.
You can find how its done [here](https://github.com/docker/metadata-action?tab=readme-ov-file#basic)

~~**Please note that the build of the `arm64` image takes a much longer time than the `amd64` one. This is due to the use of QEMU for the build process, following the Docker documentation. It may be possible to not require the use of QEMU by modifying the Dockerfile but I don't have the knowledge to do so.**~~



## To do

This PR is a Work in Progress.

- [x] Basic stuff
- [x] Test if build works (on fork)
- [x] Is opencontainers metadata correct for the images?
- [x] ~~Should the workflow use a matrix strategy for the architectures?~~ No more arm64
- [x] Remove old image building workflow?
- [x] Test running `--version` in CI
- [x] ~~More architectures? ([alpine supported architectures](https://hub.docker.com/_/alpine/tags)) Or remove `arm64`?~~ Require cross-compilation support in Dockerfile.

## How to test

Merge the branch into one of your forks of Minetest, see the workflow run, see the container in the package section of the repository. See the two architectures. Pull the image and see if it runs.

In my case:
- https://github.com/AFCMS/minetest/actions/runs/7448427624/job/20262783153
- https://github.com/AFCMS/minetest/pkgs/container/minetest
- `docker run ghcr.io/afcms/minetest:master /usr/local/bin/minetestserver --version` works
